src/addons/Portal/Portal.js

```diff
@@ -0,0 +1,255 @@
+      this.renderPortal()
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) { // eslint-disable-line complexity
+    // NOTE: Ideally the portal rendering would happen in the render() function
+    // but React gives a warning about not being pure and suggests doing it
+    // within this method.
+
+    // If the portal is open, render (or re-render) the portal and child.
+    if (this.state.open) {
+      this.renderPortal()
+    }
+
+    if (prevState.open && !this.state.open) {
+      debug('portal closed')
+      this.unmountPortal()
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmountPortal()
+  }
+
+  // ----------------------------------------
+  // Document Event Handlers
+  // ----------------------------------------
+
+  closeOnDocumentClick = (e) => {
+    if (!this.props.closeOnDocumentClick) return
+
+    debug('closeOnDocumentClick()')
+
+    e.stopPropagation()
+    this.close()
+  }
+
+  closeOnEscape = (e) => {
+    if (!this.props.closeOnEscape) return
+    if (keyboardKey.getCode(e) !== keyboardKey.Escape) return
+
+    debug('closeOnEscape()')
+
+    e.preventDefault()
+    this.close()
+  }
+
+  // ----------------------------------------
+  // Component Event Handlers
+  // ----------------------------------------
+
+  handleChildClick = (e) => {
+    debug('handleChildClick()')
+
+    // Prevent closeOnDocumentClick
+    e.nativeEvent.stopImmediatePropagation()
+  }
+
+  handleTriggerBlur = (e) => {
+    if (!this.props.closeOnTriggerBlur) return
+
+    debug('handleTriggerBlur()')
+    this.close()
+  }
+
+  handleTriggerClick = (e) => {
+    if (!this.props.openOnTriggerClick) return
+
+    debug('handleTriggerClick()')
+
+    e.stopPropagation()
+    this.open()
+  }
+
+  handleTriggerFocus = (e) => {
+    if (!this.props.openOnTriggerFocus) return
+
+    debug('handleTriggerFocus()')
+    this.open()
+  }
+
+  handleTriggerMouseLeave = (e) => {
+    if (!this.props.closeOnTriggerMouseLeave) return
+
+    debug('handleTriggerMouseLeave()')
+    this.close()
+  }
+
+  handleTriggerMouseOver = (e) => {
+    if (!this.props.openOnTriggerMouseOver) return
+
+    debug('handleTriggerMouseOver()')
+    this.open()
+  }
+
+  // ----------------------------------------
+  // Behavior
+  // ----------------------------------------
+
+  open = () => {
+    debug('open()')
+    this.trySetState({ open: true })
+  }
+
+  close = () => {
+    debug('close()')
+    this.trySetState({ open: false })
+  }
+
+  renderPortal() {
+    const { children, className } = this.props
+
+    this.mountPortal()
+
+    this.node.className = className
+
+    const child = cloneElement(Children.only(children), {
+      onClick: this.handleChildClick,
+    })
+
+    this.portal = ReactDOM.unstable_renderSubtreeIntoContainer(
+      this,
+      child,
+      this.node
+    )
+  }
+
+  mountPortal = () => {
+    if (this.node) return
+
+    this.node = document.createElement('div')
+    document.body.appendChild(this.node)
```

Comment:
I second this request, the popup might need to appear on a different component than body. Something to consider is how to handle the case where the container where to get dynamically destroyed.